feat(analytics): Implement DotCMS Analytics SDK with enhanced tracking and configuration

oidacra opened the pull request:
### Summary of Changes
- Introduced a new analytics SDK for DotCMS, including core functionalities for tracking page views and custom events for traditional DotCMS Pages.
- Added plugins for identity management and data enrichment to enhance analytics payloads.
- Implemented activity tracking for user sessions and inactivity detection.
- Updated the build process to include an analytics script and configuration options.
- Refactored existing components to utilize the new analytics SDK.

This commit enhances the analytics capabilities of DotCMS, providing a more robust and flexible tracking solution.

This PR fixes: #32858

<img width="4476" height="2352" alt="CleanShot 2025-09-04 at 14 23 01@2x" src="https://github.com/user-attachments/assets/acc93e9f-6e5c-419e-8782-b2f0e1406bd8" />

<img width="5026" height="1644" alt="CleanShot 2025-09-04 at 14 23 28@2x" src="https://github.com/user-attachments/assets/7d0f7e6e-d282-46da-9c51-f2cdc741d489" />


This PR fixes: #32858

oidacra committed:
feat(analytics): Implement DotCMS Analytics SDK with enhanced tracking and configuration

- Introduced a new analytics SDK for DotCMS, including core functionalities for tracking page views and custom events.
- Added plugins for identity management and data enrichment to enhance analytics payloads.
- Implemented activity tracking for user sessions and inactivity detection.
- Updated the build process to include an analytics script and configuration options.
- Refactored existing components to utilize the new analytics SDK.

This commit enhances the analytics capabilities of DotCMS, providing a more robust and flexible tracking solution.

oidacra committed:
git ignore changes

oidacra committed:
fix(analytics): Update build configuration and output paths for SDK

### Summary of Changes
- Removed the clean command from the build arguments for the analytics SDK.
- Updated the output path for the standalone build to ensure proper resource management.
- Added a new resource configuration to include JavaScript files from the analytics SDK in the build process.

These changes streamline the build process and enhance the integration of the analytics SDK within the project.

erickgonzalez reviewed (approved):
Java Files looks good

oidacra committed:
docs(analytics): Update README for SDK usage and configuration

### Summary of Changes
- Revised the quick start section to clarify the usage of the standalone script tag.
- Updated configuration options to reflect changes in attribute names and defaults.
- Enhanced examples for using the analytics SDK with React, including manual page view tracking.
- Added details on standalone attributes for better integration guidance.

These updates improve the documentation for developers integrating the DotCMS Analytics SDK.

oidacra committed:
fix(analytics): Update attribute names in SDK documentation and tests

### Summary of Changes
- Changed `data-server` to `data-analytics-server` in README and related code files for consistency.
- Updated configuration options in the README to reflect the new attribute names and their defaults.
- Adjusted test cases to ensure they reference the updated attribute names.

These changes enhance clarity and maintain consistency across the SDK documentation and implementation.

oidacra committed:
docs(analytics): Refine README configuration options and attribute names

### Summary of Changes
- Updated the configuration options table in the README for clarity and consistency.
- Changed the default value for `autoPageView` to specify different defaults for React and standalone usage.
- Corrected the attribute name from `data-server` to `data-analytics-server` for better alignment with SDK implementation.

These updates enhance the documentation for developers using the DotCMS Analytics SDK, ensuring accurate and clear guidance.

oidacra committed:
fix(analytics): Update import path for dot-content-analytics in tests

oidacra committed:
fix(analytics): Update site key retrieval in AnalyticsWebAPIImpl to handle missing values

### Summary of Changes
- Modified the site key retrieval logic in `AnalyticsWebAPIImpl` to return a blank string when the key is not present, improving robustness.
- Updated corresponding test cases in `AnalyticsWebAPITest` to reflect the new behavior.

These changes enhance the reliability of the analytics functionality by ensuring that missing site keys do not cause errors.

oidacra committed:
Merge branch 'main' into 32858-analytics-traditional
